commit 0cd665bd20f9088d363158b4ac75592af18ecf4f
Author: Paolo Bonzini <pbonzini@redhat.com>
Date:   Wed Mar 25 12:50:03 2020 -0400

    KVM: x86: cleanup kvm_inject_emulated_page_fault
    
    To reconstruct the kvm_mmu to be used for page fault injection, we
    can simply use fault->nested_page_fault.  This matches how
    fault->nested_page_fault is assigned in the first place by
    FNAME(walk_addr_generic).
    
    Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>

diff --git a/arch/x86/kvm/mmu/paging_tmpl.h b/arch/x86/kvm/mmu/paging_tmpl.h
index 9bdf9b7d9a96..efec7d27b8c5 100644
--- a/arch/x86/kvm/mmu/paging_tmpl.h
+++ b/arch/x86/kvm/mmu/paging_tmpl.h
@@ -812,7 +812,7 @@ static int FNAME(page_fault)(struct kvm_vcpu *vcpu, gpa_t addr, u32 error_code,
 	if (!r) {
 		pgprintk("%s: guest page fault\n", __func__);
 		if (!prefault)
-			inject_page_fault(vcpu, &walker.fault);
+			kvm_inject_emulated_page_fault(vcpu, &walker.fault);
 
 		return RET_PF_RETRY;
 	}